### PR TITLE
[rllib] test all combinations of {obs_space} x {action_space}

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Adds an exhaustive test or all combinations of supported action and obs spaces. There are some known failures (mostly in tuple action spaces).

cc @richardliaw

### CODE REVIEW

File: python/ray/rllib/models/catalog.py [hunk 1]
```diff
@@ -63,17 +63,21 @@ def get_action_dist(action_space, dist_type=None):
             dist_dim (int): The size of the input vector to the distribution.
         """
 
+        # TODO(ekl) are list spaces valid?
+        if isinstance(action_space, list):
```
Comment (ericl) [inline python/ray/rllib/models/catalog.py]: @eugenevinitsky should the action spaces for multi-agent envs be lists or Tuples?

File: .travis.yml [hunk 2]
```diff
@@ -126,6 +126,7 @@ script:
   - python test/cython_test.py
   - python -m pytest test/dataframe.py
 
+  - python -m pytest python/ray/rllib/test/test_supported_spaces.py
```
Comment (ericl) [inline .travis.yml]: Note that this takes ~200s to run

File: test/jenkins_tests/run_multi_node_tests.sh [hunk 3]
```diff
@@ -159,6 +159,9 @@ docker run --rm --shm-size=10G --memory=10G $DOCKER_SHA \
 docker run --rm --shm-size=10G --memory=10G $DOCKER_SHA \
     python /ray/python/ray/rllib/test/test_checkpoint_restore.py
 
+docker run --rm --shm-size=10G --memory=10G $DOCKER_SHA \
```
Comment (ericl) [inline test/jenkins_tests/run_multi_node_tests.sh]: takes ~200s to run

File: python/ray/rllib/models/catalog.py (hunk 1 above)
Comment (eugenevinitsky) [reply python/ray/rllib/models/catalog.py]: The action spaces are lists, the observation spaces are tuples. The observation space was to take advantage of the tuple flattening preprocessor. This isn't very consistent so I'll submit a pull request fixing this.

Comment (ericl) [reply python/ray/rllib/models/catalog.py]: Ok cool, feel free to remove this conversion later if you think just supporting Tuple is sufficient.

File: python/ray/rllib/test/test_supported_spaces.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            "ES",
+            {"num_workers": 1, "noise_size": 10000000,
+             "episodes_per_batch": 1, "timesteps_per_batch": 1},
+            stats)
+        num_unexpected_errors = 0
+        num_unexpected_success = 0
+        for (alg, a_name, o_name), stat in sorted(stats.items()):
+            if stat in ["ok", "unsupported"]:
+                if (alg, a_name, o_name) in KNOWN_FAILURES:
+                    num_unexpected_success += 1
```
Comment (richardliaw) [inline python/ray/rllib/test/test_supported_spaces.py]: why is this dual counting necessary? why not just assert `stat not in ["ok", "unsupported"]`?

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
mac tests failing due to logs too long (and Linux tests pass)  - merging

**Pull request merged**